src/link/MachO.zig

```diff
@@ -843,7 +838,11 @@ pub fn resolveLibSystem(
         // re-exports every single symbol definition.
         for (search_dirs) |dir| {
             if (try resolveLib(arena, dir, "System", ".tbd")) |full_path| {
-                try out_libs.put(full_path, .{ .needed = true });
+                try out_libs.put(full_path, .{
```

Comment:
Perhaps not in this PR (I can follow up with a subsequent PR) but we could move this logic to `main.zig` too? This way we would have all path handling done in the frontend.